commit 44354e008984659ecc399f262e6a84a270b0a3be
Author: Dan van der Ster <daniel.vanderster@cern.ch>
Date:   Mon Apr 10 17:29:05 2017 +0200

    ceph-mgr: reopen logs on SIGHUP
    
    Wire up the sighup_handler to reopen logs, like the other daemons.
    
    Fixes: http://tracker.ceph.com/issues/19568
    Signed-off-by: Dan van der Ster <daniel.vanderster@cern.ch>

diff --git a/src/mgr/MgrStandby.cc b/src/mgr/MgrStandby.cc
index 843ecd0a3a0..af176fd9b24 100644
--- a/src/mgr/MgrStandby.cc
+++ b/src/mgr/MgrStandby.cc
@@ -303,12 +303,14 @@ int MgrStandby::main(vector<const char *> args)
   // Enable signal handlers
   signal_mgr = this;
   init_async_signal_handler();
+  register_async_signal_handler(SIGHUP, sighup_handler);
   register_async_signal_handler_oneshot(SIGINT, handle_mgr_signal);
   register_async_signal_handler_oneshot(SIGTERM, handle_mgr_signal);
 
   client_messenger->wait();
 
   // Disable signal handlers
+  unregister_async_signal_handler(SIGHUP, sighup_handler);
   unregister_async_signal_handler(SIGINT, handle_mgr_signal);
   unregister_async_signal_handler(SIGTERM, handle_mgr_signal);
   shutdown_async_signal_handler();
